tests/integration_tests/test_sequence_features.py

```diff
@@ -0,0 +1,350 @@
+    ]
+
+    # generate synthetic data set testing
+    dataset = build_synthetic_dataset(
+        150,
+        copy.deepcopy(input_features) + copy.deepcopy(output_features)
+    )
+    raw_data = '\n'.join([r[0] + ',' + r[1] for r in dataset])
+    df = pd.read_csv(StringIO(raw_data))
+
+    return df, input_features, output_features
+
+# setups up minimal number of data structures required to support initialized
+# input and output features.  The function returns initialized LudwigModel
+# and batcher for training dataset
+def setup_model_scaffolding(
+    raw_df,
+    input_features,
+    output_features
+):
+
+    # setup input feature for testing
+    config = {'input_features': input_features,
+              'output_features': output_features}
+
+    # setup model scaffolding to for testing
+    model = LudwigModel(config)
+    training_set, _, _, training_set_metadata = preprocess_for_training(
+        config,
+        training_set=raw_df,
+        skip_save_processed_input=True
+    )
+    model.training_set_metadata = training_set_metadata
+    update_config_with_metadata(
+        model.config,
+        training_set_metadata
+    )
+    model.model = model.create_model(model.config)
+
+    # setup batcher to go through synthetic data
+    batcher = initialize_batcher(
+        training_set
+    )
+
+    return model, batcher
+
+# test sequence input features
+# pytest parameters:
+#   enc_cell_type: encoder cell types
+#   enc_encoder: sequence input feature encoder
+@pytest.mark.parametrize('enc_cell_type', ['lstm', 'rnn', 'gru'])
+@pytest.mark.parametrize('enc_encoder', ENCODERS)
+def test_sequence_encoders(
+    enc_encoder,
+    enc_cell_type,
+    generate_sequence_training_data
+):
+    # retrieve pre-computed dataset and features
+    raw_df = generate_sequence_training_data[0]
+    input_features = generate_sequence_training_data[1]
+    output_features = generate_sequence_training_data[2]
+    input_feature_name = input_features[0]['name']
+    input_features[0]['encoder'] = enc_encoder
+    input_features[0]['cell_type'] = enc_cell_type
+
+
+    model, batcher = setup_model_scaffolding(
+        raw_df,
+        input_features,
+        output_features
+    )
+
+    while not batcher.last_batch():
+        batch = batcher.next_batch()
+        inputs = {
+            i_feat.feature_name: batch[i_feat.feature_name]
+            for i_feat in model.model.input_features.values()
+        }
+
+        # retrieve encoder to test
+        encoder = model.model.input_features[input_feature_name].encoder_obj
+        encoder_out = encoder(tf.cast(inputs[input_feature_name], dtype=tf.int32))
+
+        # check encoder output for proper content, type and shape
+        batch_size = batch[input_feature_name].shape[0]
+        seq_size = input_features[0]['max_len']
+
+        assert 'encoder_output' in encoder_out
+        assert isinstance(encoder_out['encoder_output'], tf.Tensor)
+
+        if enc_encoder == 'parallel_cnn':
```

Comment:
one general consideration about the shapes of the outputs in all those if cases: the rank of the encoder_output  tensor depends on the reduce_output parameter, while encoder_output_state is not affected by it. not sure if we want to test for this too or not, but wnated to point it out if needed

Reply:
I understand the point.   I know at this time the `reduce_output` and `reduce_input` are specified as `None`.   Over time, we should the add other parameters for testing the encoders/decoders and adjust the assertion test to account for those parameters.

I wanted to show at least one approach for setting up unit tests specifically for encoders and decoders.  Does this approach seem reasonable?

Reply:
Yes the approach is definitely reasonable. I was pointing this out just so we know there are potentially other parameters that we are not testing for at the moment that influence the shape ;)